bundles/org.eclipse.wst.xml.xpath2.processor/src/org/eclipse/wst/xml/xpath2/processor/internal/types/XSAnyURI.java

```diff
@@ -107,7 +110,17 @@ public ResultSequence constructor(ResultSequence arg) throws DynamicError {
 			throw DynamicError.invalidType();
 		}
 
-		return new XSAnyURI(aat.string_value().trim());
+		String uri = aat.string_value().trim().replaceAll("\\s+", " ");
+
+		try {
+			// The URI constructor does not allow raw spaces. Escape them to ensure they do not cause the xs:anyURI
+			// constructor to fail.
+			new URI(uri.replace(" ", "%20"));
```

Comment:
It does appear that this would not be sufficient. The `xs:anyURI` type actually specifies the escaping algorithm to run prior to checking if the result is a valid URI syntax.

From [XML Schema Part 2: Datatypes Second Edition §3.2.17 anyURI](https://www.w3.org/TR/xmlschema-2/#anyURI):

> The ·lexical space· of **anyURI** is finite-length character sequences which, when the algorithm defined in Section 5.4 of [XML Linking Language] is applied to them, result in strings which are legal URIs according to [RFC 2396], as amended by [RFC 2732].

From [XML Linking Language (XLink) Version 1.0 §5.4 Locator Attribute (`href`)](https://www.w3.org/TR/2001/REC-xlink-20010627/#link-locators):

> Some characters are disallowed in URI references, even if they are allowed in XML; the disallowed characters include all non-ASCII characters, plus the excluded characters listed in Section 2.4 of [IETF RFC 2396], except for the number sign (#) and percent sign (%) and the square bracket characters re-allowed in [IETF RFC 2732]. Disallowed characters must be escaped as follows:
>
> 1. Each disallowed character is converted to UTF-8 [IETF RFC 2279] as one or more bytes.
> 2. Any bytes corresponding to a disallowed character are escaped with the URI escaping mechanism (that is, converted to *%HH*, where HH is the hexadecimal notation of the byte value).
> 3. The original character is replaced by the resulting character sequence.